src/deleteInWithCleanUp.js

```diff
@@ -9,7 +9,12 @@ const createDeleteInWithCleanUp = ({ deepEqual, empty, getIn, deleteIn, setIn })
       const parent = getIn(state, pathTokens.join('.'))
       return parent ? setIn(state, path, undefined) : state
     }
-    const result = deleteIn(state, path)
+
+    let result = state
+    if (typeof getIn(state, path) !== 'undefined') {
```

Comment:
`typeof myvar != "undefined"` is usually only useful for checking if global vars exist - for vars you know exist, use simply `myvar !== undefined`.

Not a big deal here, though, so no need to bother changing it!